packages/core/strapi/lib/commands/actions/develop/action.js

```diff
@@ -106,12 +106,33 @@ const primaryProcess = async ({ distDir, appDir, build, isTSProject, watchAdmin,
   cluster.fork();
 };
 
-const workerProcess = ({ appDir, distDir, watchAdmin, polling, isTSProject }) => {
-  const strapiInstance = strapi({
+const workerProcess = async ({ appDir, distDir, watchAdmin, polling, isTSProject }) => {
+  const strapiInstance = await strapi({
     distDir,
     autoReload: true,
     serveAdminPanel: !watchAdmin,
-  });
+  }).load();
+
+  /**
+   * TypeScript automatic type generation upon dev server restart
+   * Its implementation, configuration and behavior can change in future releases
+   * @experimental
+   */
+  const shouldGenerateTypeScriptTypes = strapiInstance.config.get(
+    'server.typescript.dev.generate',
```

Comment:
```suggestion
    'server.typescript.autogenerate',
```
How about something like this? I'd rather keep it flat if possible instead of grouping it under dev, and `generate` also isn't explanatory that it's automatic generation.

Reply:
My only concern is that it would be lacking the context about the automatic generation being only for dev mode

Reply:
But I do agree with your comment about generate vs autogenerate. I think my preferred option would be `server.typescript.dev.autogenerate`

Reply:
I'm trying to understand the best place to put it from the docs. I think config/server is intended to only configure the servers, whereas config/admin is used for things like the develop watchers (for example, it has `watchIgnoreFiles`)

Admin: https://docs.strapi.io/dev-docs/configurations/admin-panel
Server: https://docs.strapi.io/dev-docs/configurations/server

What do you think about `admin.typescript.watchContentSchemas` or something like that?

Reply:
admin config options should only contain those related to the admin panel (in which case we should reload the admin, on which port should we server it, etc...) it's not meant to configure options for the server or other tools that doesn't rely on the admin

Reply:
What we could do is have a new typescript config file :eyes:

Reply:
I've updated the PR description to use `typescript.autogenerate` instead of `server.typescript.dev.autogenerate`